File: codes/scala/chapter_hashing/built_in_hash.scala

import scala.collection.mutable.ArrayBuffer

/**
 * File: built_in_hash.scala
 * Author: WangBlue (wangjiping596@gmail.com)
 * Create: 2024-08-12 16:03
 */

object built_in_hash {
  def main(args: Array[String]): Unit = {
    // 整数的哈希值
    val num = 3
    val hashNum = num.hashCode
    println(s"整数 $num 的哈希值为 $hashNum")

    // 布尔量的哈希值
    val bol = true
    val hashBol = bol.hashCode
    println(s"布尔量 $bol 的哈希值为 $hashBol")

    // 小数的哈希值
    val dec = 3.14159
    val hashDec = dec.hashCode
    println(s"小数 $dec 的哈希值为 $hashDec")

    // 字符串的哈希值
    val str = "Hello 算法"
    val hashStr = str.hashCode
    println(s"字符串 $str 的哈希值为 $hashStr")

    // 数组的哈希值
    val arr = Array(12836, "小哈")
    val hashTup = arr.hashCode
    println(s"数组 ${arr.mkString("[", ", ", "]")} 的哈希值为 $hashTup")


    val obj = new ListNode(0)
    val hashObj = obj.hashCode
    println(s"节点对象 $obj 的哈希值为 $hashObj")

  }

}

File: codes/java/chapter_hashing/built_in_hash.java
/**
 * File: built_in_hash.java
 * Created Time: 2023-06-21
 * Author: krahets (krahets@163.com)
 */

package chapter_hashing;

import utils.*;
import java.util.*;

public class built_in_hash {
    public static void main(String[] args) {
        int num = 3;
        int hashNum = Integer.hashCode(num);
        System.out.println("整数 " + num + " 的哈希值为 " + hashNum);

        boolean bol = true;
        int hashBol = Boolean.hashCode(bol);
        System.out.println("布尔量 " + bol + " 的哈希值为 " + hashBol);

        double dec = 3.14159;
        int hashDec = Double.hashCode(dec);
        System.out.println("小数 " + dec + " 的哈希值为 " + hashDec);

        String str = "Hello 算法";
        int hashStr = str.hashCode();
        System.out.println("字符串 " + str + " 的哈希值为 " + hashStr);

        Object[] arr = { 12836, "小哈" };
        int hashTup = Arrays.hashCode(arr);
        System.out.println("数组 " + Arrays.toString(arr) + " 的哈希值为 " + hashTup);

        ListNode obj = new ListNode(0);
        int hashObj = obj.hashCode();
        System.out.println("节点对象 " + obj + " 的哈希值为 " + hashObj);
    }
}
